yt/yt/server/query_tracker/query_tracker.cpp

```diff
@@ -646,6 +670,24 @@ class TQueryTracker
                 YT_LOG_ERROR(commitResultOrError, "Failed to finish query, backing off");
                 return true;
             }
+
+            // Save profile counter
+            auto tags = TProfilingTags{
+                .State = activeQueryRecord->State,
+                .Engine = activeQueryRecord->Engine,
+                .AssignedTracker = activeQueryRecord->AssignedTracker.value_or(NoneQueryTracker),
+            };
+            {
+                auto guard = std::make_optional(Guard(StateTimeProfilingCountersMap_->Lock));
+                auto& stateTimeGauge = GetOrCreateProfilingCounter(
+                    QueryTrackerProfiler,
+                    tags,
+                    StateTimeProfilingCountersMap_->Map,
+                    guard).StateTime;
+                auto now = TInstant::Now();
```

Comment:
If you are talking about this loop, it is useful to recalculate it every iteration because they are too long.
https://github.com/ytsaurus/ytsaurus/blob/b2952c4fd20b002525efd1c862e6f6b264fb954a/yt/yt/server/query_tracker/query_tracker.cpp#L532